test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -1,48 +1,73 @@
 
 	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
 	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/cluster"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/discovery"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/images"
+	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
-var workerRTNode *corev1.Node
-var profile *performancev2.PerformanceProfile
-var RunningOnSingleNode bool
+var (
+	workerRTNode            *corev1.Node
+	workerRTNodes           []corev1.Node
+	profile, initialProfile *performancev2.PerformanceProfile
+	mc                      *machineconfigv1.MachineConfig
+	RunningOnSingleNode     bool
+	activation_file         string = "/rootfs/var/lib/ovn-ic/etc/enable_dynamic_cpu_affinity"
+	performanceMCP          string
+)
 
-var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+var s_ = Describe("[performance] Cgroups and affinity", Ordered, func() {
 	var onlineCPUSet cpuset.CPUSet
 
-	testutils.CustomBeforeAll(func() {
-		isSNO, err := cluster.IsSingleNode()
-		Expect(err).ToNot(HaveOccurred())
-		RunningOnSingleNode = isSNO
-	})
-
-	BeforeEach(func() {
-		if discovery.Enabled() && testutils.ProfileNotFound {
-			Skip("Discovery mode enabled, performance profile not found")
-		}
-
+	BeforeAll(func() {
 		workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
 		Expect(err).ToNot(HaveOccurred())
+
 		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
 		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("error looking for the optional selector: %v", err))
-		Expect(workerRTNodes).ToNot(BeEmpty())
 		workerRTNode = &workerRTNodes[0]
+
 		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
 		Expect(err).ToNot(HaveOccurred())
 
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+	})
+
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
```

Comment:
This check can move to `BeforeAll` as well

Reply:
Addressed this in the latest commit

Reply:
Why do we still check it in `BeforeEach`?

Reply:
Addressed in the latest commit